src/core/ddsi/src/ddsi__tran.h

```diff
@@ -252,15 +252,15 @@ inline dds_return_t ddsi_factory_create_conn (struct ddsi_tran_conn **conn, stru
   *conn = NULL;
   if ((qos->m_interface != NULL) != (qos->m_purpose == DDSI_TRAN_QOS_XMIT_UC || qos->m_purpose == DDSI_TRAN_QOS_XMIT_MC))
     return DDS_RETCODE_BAD_PARAMETER;
-  if (!ddsi_is_valid_port (factory, port))
+  if (port != 0 && !ddsi_is_valid_port (factory, port))
```

Comment:
This condition (and the ones below) confused me a bit, because port number `0` was rejected earlier in the check in `do_locator`. Because at this point the `0` has a special meaning for the underlying implementation, I think it would be more clear to replace it with a #define with a descriptive name.